### PR TITLE
feat: add error event for utility process

### PR DESCRIPTION
**Pull request by deepak1556**
#### Description of Change

This enables application to collect additional diagnostic data for telemetry purposes. Primary use case for VSCode is to identify addons loaded by 3rd party extensions in the utility process that could trigger V8 Fatal error due to sandbox violation. Additional details available at https://gist.github.com/deepak1556/ec4a882156855f0d7870a4a7c01758e3

The error event can be extended in future on a need basis to expose other error types that the Node.js report module supports.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Add error event in utility process to support diagnostic reports on V8 fatal errors

### CODE REVIEW

File: patches/node/feat_expose_report_writing_entrypoint_to_embedders.patch [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+-static void WriteNodeReport(Isolate* isolate,
+-                            Environment* env,
+-                            const char* message,
+-                            const char* trigger,
+-                            const std::string& filename,
+-                            std::ostream& out,
+-                            Local<Value> error,
+-                            bool compact,
+-                            bool exclude_network) {
++void WriteNodeReport(Isolate* isolate,
```
Comment (codebytere) [inline patches/node/feat_expose_report_writing_entrypoint_to_embedders.patch:42]: @deepak1556 have you attempted to upstream this to Node.js?

Comment (deepak1556) [reply patches/node/feat_expose_report_writing_entrypoint_to_embedders.patch:42]: Not yet

Comment (deepak1556) [reply patches/node/feat_expose_report_writing_entrypoint_to_embedders.patch:42]: Patch has been removed in favor of the public API

**Review by codebytere — APPROVED** [review]
API LGTM

File: docs/api/utility-process.md [hunk 2]
```diff
@@ -116,6 +116,17 @@ When the child process exits, then the value is `null` after the `exit` event is
...
+
+Returns:
+
+* `type` string - Type of error. One of the following values:
+  * `FatalError`
+* `location` string - Source location from where the error originated.
+* `report` string - [`Node.js diagnostic report`][].
+
+Emitted when the child process needs to terminate due to non continuable error from V8.
+
```
[the remark this answers is not in the document]
Comment (deepak1556) [reply docs/api/utility-process.md:129]: Good idea, will add that.

[the remark this answers is not in the document]
Comment (deepak1556) [reply docs/api/utility-process.md]: Updated, PTAL.

**Review by jkleinsc — APPROVED** [review]

**Review by ckerr — APPROVED** [review]

**Pull request merged**